src/Microsoft.ML.PipelineInference/AutoInference.cs

```diff
@@ -56,13 +56,13 @@ public class DependencyMap : Dictionary<int, LevelDependencyMap> { }
         /// </summary>
         public sealed class SupportedMetric
         {
-            public static readonly SupportedMetric Auc = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.Auc, true);
+            public static readonly SupportedMetric Auc = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.Auc.ToUpper(), true);
             public static readonly SupportedMetric AccuracyMicro = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.AccuracyMicro, true);
             public static readonly SupportedMetric AccuracyMacro = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.AccuracyMacro, true);
             public static readonly SupportedMetric L1 = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.L1, false);
             public static readonly SupportedMetric L2 = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.L2, false);
             public static readonly SupportedMetric F1 = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.F1, true);
-            public static readonly SupportedMetric AuPrc = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.AuPrc, true);
+            public static readonly SupportedMetric AuPrc = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.AuPrc.ToUpper(), true);
             public static readonly SupportedMetric TopKAccuracy = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.TopKAccuracy, true);
             public static readonly SupportedMetric Rms = new SupportedMetric(FieldNames.PipelineSweeperSupportedMetrics.Rms, false);
```

Comment:
>Rms [](start = 120, length = 3)

This?

Reply:
I'm not sure of a good way to do this, but am open to suggestions. These show up as all caps in the metrics IDV's, which was causing a failure (since it was looking for "Auc", the name given, rather than "AUC", what it was in the IDV). The failure wasn't seen previously, since a bug was causing the first column to always get picked up (which coincidentally was the AUC column), so fixing the bug exposed this problem. We want the name of the supported metric to match what it will be in the IDV when looking to extract it. Hence the upper case. Should I just hard-code to the string "AUC" / "AUPRC" here instead?

Reply:
I'm not sure. So you are saying that *something* is producing columns with `AUC` in the name, but we were looking for `Auc` because `FieldNames.PipelineSweeperSupportedMetrics` defines it that way, and you fix it this way?

Is there a reason why we would not just either fix `FieldNames.PipelineSweeperSupportedMetrics` to be `AUC`, or fix whatever is producing a data view with column `AUC` to produce `Auc`?

That is: my impression is that this seems like a bandaid on a more serious inconsistency between two pieces of code, is that impression accurate or am I wrong? If I'm right I'd rather fix that inconsistency.

---
In reply to: [189666725](https://github.com/dotnet/machinelearning/pull/152#discussion_r189666725) [](ancestors = 189666725)

Reply:
The FieldNames.PipelineSweeperSupportedMetrics string is defined to be whatever we want. Since these are shown to the user (via the GUI, for example), I was wary of changing them to be AUC, etc., unilaterally. So the change is easy (I can just change them there) if no one has a problem with having them all caps. (It is weird to worry about capitalization, but that is also a big deal in the codebase, getting capitalization right and consistent). With this way, I was only changing it on my end, where I'd be using it, so it seemed like the least intrusive of the options. But your reply makes me think it wasn't the way to go.